src/python/pants/option/compiler_option_sets_mixin.py

```diff
@@ -62,16 +62,23 @@ def get_merged_args_for_compiler_option_sets(self, compiler_option_sets):
       compiler_option_sets = self.get_options().default_compiler_option_sets
       logger.debug('using default option sets: {}'.format(compiler_option_sets))
 
-    compiler_options = set()
+    compiler_options = []
```

Comment:
This feature could really use some docs, not sure who originally added the feature. I struggled to trace the logic when I tripped over it a few months ago.

Not a blocking issue for your PR, just floating a casual feature request in case the original author is reading. 🙈

Reply:
It was added at the very end of an intern rotation last year (for which we were very grateful), but as a result didn't get the depth of initial review it probably should have. This could definitely be better-documented, especially as its semantics are being fixed here, and a related-but-different feature recently just changed again in #7594. I'll make a followup issue/PR for adding a section to https://www.pantsbuild.org/options.html on this I think?